snabbdom/src/test/scala/com/github/buntec/snabbdom/AttributesSuite.scala

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2022 buntec
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Copyright (c) 2015 Simon Friis Vindum
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+package com.github.buntec.snabbdom
+
+import org.scalajs.dom
+
+class AttributesSuite extends BaseSuite {
+
+  val vnode0 = FunFixture[dom.Element](
+    setup = { _ =>
+      dom.document.createElement("div")
+    },
+    teardown = { _ => () }
+  )
+
+  val patch = init(Seq(modules.Attributes.module))
+
+  group("attributes") {
+
+    vnode0.test("have their provided values") { vnode0 =>
+      val vnode1 = h(
+        "div",
+        VNodeData.builder
+          .withAttrs(
+            "href" -> "/foo",
+            "minlength" -> 1,
+            "selected" -> true,
+            "disabled" -> false
```

Comment:
So @davesmith00000 brought this up as well, should attributes just be set directly as `String`? Since that is their canonical form, and otherwise you don't get round-tripping with `getAttribute()`.

Reply:
Sounds reasonable to me!

Reply:
But then what about the special treatment of boolean attributes?

Reply:
Yeah, what exactly is the special treatment? Could we just use `Option[String]` with `Some("")` and `None`?